core/validator.go

```diff
@@ -43,74 +108,132 @@ func (v *chunkValidator) ValidateBlob(blob *BlobMessage, operatorState *Operator
 	}
 
 	for _, quorumHeader := range blob.BlobHeader.QuorumInfos {
-
-		if quorumHeader.AdversaryThreshold >= quorumHeader.QuorumThreshold {
-			return errors.New("invalid header: quorum threshold does not exceed adversary threshold")
-		}
-
-		// Check if the operator is a member of the quorum
-		if _, ok := operatorState.Operators[quorumHeader.QuorumID]; !ok {
+		// preprocess validation info
+		chunks, assignment, params, err := v.preprocessBlob(quorumHeader, blob, operatorState)
+		if err == ErrBlobQuorumSkip {
 			continue
-		}
-
-		// Get the assignments for the quorum
-		assignment, info, err := v.assignment.GetOperatorAssignment(operatorState, quorumHeader.QuorumID, quorumHeader.QuantizationFactor, v.operatorID)
-		if err != nil {
+		} else if err != nil {
 			return err
+		} else {
+			// Check the received chunks against the commitment
+			err = v.encoder.VerifyChunks(chunks, assignment.GetIndices(), blob.BlobHeader.BlobCommitments, *params)
+			if err != nil {
+				return err
+			}
 		}
+	}
 
-		// Validate the number of chunks
-		if assignment.NumChunks == 0 {
-			continue
-		}
-		if assignment.NumChunks != uint(len(blob.Bundles[quorumHeader.QuorumID])) {
-			return errors.New("number of chunks does not match assignment")
-		}
+	return nil
+}
 
-		chunkLength, err := v.assignment.GetChunkLengthFromHeader(operatorState, quorumHeader)
-		if err != nil {
-			return err
-		}
+func (v *chunkValidator) UpdateOperatorID(operatorID OperatorID) {
+	v.operatorID = operatorID
+}
 
-		// Validate the chunkLength against the quorum and adversary threshold parameters
-		numOperators := uint(len(operatorState.Operators[quorumHeader.QuorumID]))
-		minChunkLength, err := v.assignment.GetMinimumChunkLength(numOperators, blob.BlobHeader.BlobCommitments.Length, quorumHeader.QuantizationFactor, quorumHeader.QuorumThreshold, quorumHeader.AdversaryThreshold)
-		if err != nil {
-			return err
-		}
-		params, err := GetEncodingParams(minChunkLength, info.TotalChunks)
-		if err != nil {
-			return err
-		}
+func (v *chunkValidator) ValidateBatch(blobs []*BlobMessage, operatorState *OperatorState, pool common.WorkerPool) error {
+	subBatchMap := make(map[EncodingParams]*SubBatch)
+	blobCommitmentList := make([]BlobCommitments, len(blobs))
 
-		if params.ChunkLength != chunkLength {
-			return errors.New("number of chunks does not match assignment")
+	for k, blob := range blobs {
+		if len(blob.Bundles) != len(blob.BlobHeader.QuorumInfos) {
+			return errors.New("number of bundles does not match number of quorums")
 		}
 
-		// Get the chunk length
-		chunks := blob.Bundles[quorumHeader.QuorumID]
-		for _, chunk := range chunks {
-			if uint(chunk.Length()) != chunkLength {
-				return ErrChunkLengthMismatch
+		// Validate the blob length
+
+		blobCommitmentList[k] = blob.BlobHeader.BlobCommitments
+
+		// for each quorum
+		for _, quorumHeader := range blob.BlobHeader.QuorumInfos {
+			// Check if the operator is a member of the quorum
```

Comment:
nit: this comment is redundant (it's in preprocessBlob where the check is actually happening)